@afc163 Looks like the tests are unstable. I did not touch any code regarding the Form component.
```
2022-09-22T20:57:56.0496231Z Summary of all failing tests
2022-09-22T20:57:56.0501066Z FAIL components/form/__tests__/index.test.tsx (17.201 s)
2022-09-22T20:57:56.0504038Z   ● Form › Form.Item.useStatus should work
2022-09-22T20:57:56.0504401Z 
2022-09-22T20:57:56.0506812Z     expect(received).toContain(expected) // indexOf
2022-09-22T20:57:56.0509152Z 
2022-09-22T20:57:56.0509474Z     Expected value:  "custom-input-status-error"
2022-09-22T20:57:56.0512476Z     Received object: {"0": "custom-input-required", "1": "custom-input-status-"}
2022-09-22T20:57:56.0513632Z 
2022-09-22T20:57:56.0513638Z 
2022-09-22T20:57:56.0513643Z 
2022-09-22T20:57:56.0515965Z       at _callee24$ (components/form/__tests__/index.test.tsx:1826:191)
2022-09-22T20:57:56.0518675Z       at call (node_modules/@babel/runtime/helpers/regeneratorRuntime.js:86:17)
2022-09-22T20:57:56.0521891Z       at Generator.tryCatch [as _invoke] (node_modules/@babel/runtime/helpers/regeneratorRuntime.js:66:24)
2022-09-22T20:57:56.0524893Z       at Generator._invoke [as next] (node_modules/@babel/runtime/helpers/regeneratorRuntime.js:117:21)
2022-09-22T20:57:56.0527709Z       at asyncGeneratorStep (node_modules/@babel/runtime/helpers/asyncToGenerator.js:3:16)
2022-09-22T20:57:56.0530198Z       at asyncGeneratorStep (node_modules/@babel/runtime/helpers/asyncToGenerator.js:25:9)
2022-09-22T20:57:56.0530404Z 
2022-09-22T20:57:56.0531615Z 
2022-09-22T20:57:56.0535052Z Test Suites: 1 failed, 148 passed, 149 total
2022-09-22T20:57:56.0535334Z Tests:       1 failed, 12 skipped, 1663 passed, 1676 total
2022-09-22T20:57:56.0535578Z Snapshots:   895 passed, 895 total
2022-09-22T20:57:56.0535786Z Time:        213.213 s
2022-09-22T20:57:56.0535978Z Ran all test suites.
2022-09-22T20:57:56.0785177Z ##[error]Process completed with exit code 1.
2022-09-22T20:57:56.0859230Z Post job cleanup.
2022-09-22T20:57:56.2098241Z [command]/usr/bin/git version
2022-09-22T20:57:56.2158026Z git version 2.37.3
```

--- a/components/notification/__tests__/index.test.tsx
+++ b/components/notification/__tests__/index.test.tsx
@@ -336,4 +336,16 @@ describe('notification', () => {
 
     expect(document.querySelectorAll('.anticon-user').length).toBe(1);
   });
+
+  it('support props', () => {
+    act(() => {
+      notification.open({
+        message: 'Notification Title',
+        duration: 0,
+        props: { 'data-testid': 'test-notification' },
+      });
+    });
+
+    expect(document.querySelectorAll("[data-testid='test-notification']").length).toBe(1);
+  });
 });

--- a/components/notification/index.en-US.md
+++ b/components/notification/index.en-US.md
@@ -46,6 +46,7 @@ The properties of config are as follows:
 | top | Distance from the top of the viewport, when `placement` is `topRight` or `topLeft` (unit: pixels) | number | 24 |
 | onClick | Specify a function that will be called when the notification is clicked | function | - |
 | onClose | Trigger when notification closed | function | - |
+| props | Props passed down | Object | An object that can contain `data-*`, `aria-*`, or `role` props, to be put on the notification `div`. This currently only allows `data-testid` instead of `data-*` in TypeScript. See https://github.com/microsoft/TypeScript/issues/28960. |
 
 `notification` also provides a global `config()` method that can be used for specifying the default options. Once this method is used, all the notification boxes will take into account these globally defined options when displaying.
 

--- a/components/notification/index.tsx
+++ b/components/notification/index.tsx
@@ -10,6 +10,10 @@ import * as React from 'react';
 import ConfigProvider, { globalConfig } from '../config-provider';
 import createUseNotification from './hooks/useNotification';
 
+interface DivProps extends React.HTMLProps<HTMLDivElement> {
+  'data-testid'?: string;
+}
+
 export type NotificationPlacement =
   | 'top'
   | 'topLeft'
@@ -43,6 +47,7 @@ export interface ConfigProps {
   closeIcon?: React.ReactNode;
   rtl?: boolean;
   maxCount?: number;
+  props?: DivProps;
 }
 
 function setNotificationConfig(options: ConfigProps) {
@@ -216,6 +221,7 @@ export interface ArgsProps {
   bottom?: number;
   getContainer?: () => HTMLElement;
   closeIcon?: React.ReactNode;
+  props?: DivProps;
 }
 
 function getRCNoticeProps(args: ArgsProps, prefixCls: string, iconPrefixCls?: string) {
@@ -232,6 +238,7 @@ function getRCNoticeProps(args: ArgsProps, prefixCls: string, iconPrefixCls?: st
     style,
     className,
     closeIcon = defaultCloseIcon,
+    props,
   } = args;
 
   const duration = durationArg === undefined ? defaultDuration : durationArg;
@@ -280,6 +287,7 @@ function getRCNoticeProps(args: ArgsProps, prefixCls: string, iconPrefixCls?: st
     className: classNames(className, {
       [`${prefixCls}-${type}`]: !!type,
     }),
+    props,
   };
 }
 

